src/events/messageCreate/checkAvailability.js

```diff
@@ -0,0 +1,207 @@
+                        'items':       {
+                            'monday': {
+                                'type':        'bool',
+                                'description': 'If monday is available or not.',
+                            },
+                            'tuesday': {
+                                'type':        'bool',
+                                'description': 'If tuesday is available or not.',
+                            },
+                            'wednesday': {
+                                'type':        'bool',
+                                'description': 'If wednesday is available or not.',
+                            },
+                            'thursday': {
+                                'type':        'bool',
+                                'description': 'If thursday is available or not.',
+                            },
+                            'friday': {
+                                'type':        'bool',
+                                'description': 'If friday is available or not.',
+                            }
+                        }
+
+                    }
+                },
+                'required': ['from', 'to', 'days'],
+            },
+        },
+    },
+    {
+        'type':     'function',
+        'function': {
+            'name':        'unableToParse',
+            'description': 'If the user inputs content that cannot be used in the other functions.',
+            'parameters':  {},
+        },
+    },
+];
+
+/**
+ * Given a function object from open ai and a message, attempt to parse 
+ * @param {DiscordJS Message} message 
+ * @param {OpenAI Tool} calledFunction 
+ */
+const parseResults = (message, calledFunction) =>{
+    const action = {
+        'rememberUnavailability': () => rememberUnavailability(message, JSON.parse(calledFunction.arguments)),
+        'rememberAvailability':   () => rememberAvailability(message, JSON.parse(calledFunction.arguments)),
+        'unableToParse':          () => unableToParse(message),
+    };
+    action[calledFunction.name]();
+};
+
+// checks the message sent in the channel and sends it to openai to parse, then if possible saves the availability data
+module.exports = async (client, message) => {
+    try {
+        const availabilityChannel = await getAvailabilityChannel();
+
+        // only send a message if it's not from a bot and it's from the available channel
+        if (message.author.bot || !availabilityChannel || message.channelId !== availabilityChannel.id) {
+            return;
+        }
+
+        const date = new Date(message.createdTimestamp).toLocaleDateString();
+        const response = await openAiClient.chat.completions.create({
+            model:    'gpt-3.5-turbo',
+            messages: [
+                {
+                    'role':    'user',
+                    'content': message.content + `the current local date is ${date} and tomorrow is not today.`,
```

Comment:
Can you include an example of the prompt being sent to openai? In full. Basically just log this, and add as a comment here.

Asking because I think I'm missing something about how this operates.

Reply:
![image](https://github.com/RIT-Crowd-Comic/Bot/assets/89936632/f01c71eb-8740-42a3-a103-7883b0901a54)

Reply:
In terms of a unit test(s) would it be essential mocking the data that openAi outputs with some set prompts. I don't have any experience writing them, and it would be valuable to learn, I just don't know how long it would take.